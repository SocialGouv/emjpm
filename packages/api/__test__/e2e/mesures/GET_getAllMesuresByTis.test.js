//

const request = require("supertest");

const { knex } = global;
jest.setMock("@emjpm/api/db/knex", knex);

const server = require("@emjpm/api/app");

const { getTokenByUserType } = require("../utils");

//

beforeAll(async () => {
  await knex.migrate.latest();
  await knex.seed.run();
});

test("should return mesures of user type prepose", async () => {
  const token = await getTokenByUserType("ti");
  const response = await request(server)
    .get("/api/v1/mesures/getAllMesuresByTis")
    .set("Authorization", "Bearer " + token);

  expect(response.body).toMatchSnapshot();

  expect(response.status).toBe(200);
});
