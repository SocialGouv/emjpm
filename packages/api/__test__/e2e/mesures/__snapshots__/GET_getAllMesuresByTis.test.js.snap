// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return mesures of user type prepose 1`] = `
Array [
  Object {
    "annee": "2010-10-05",
    "antenne_id": null,
    "cabinet": null,
    "civilite": "F",
    "code_postal": "93000",
    "created_at": "2010-10-05T00:00:00.000Z",
    "date_ouverture": "2010-10-04T22:00:00.000Z",
    "department_id": null,
    "etablissement": "UDAHF",
    "etablissement_id": null,
    "extinction": null,
    "id": 3,
    "manda": "UDAHF",
    "mandataire_id": 1,
    "nom": null,
    "numero_dossier": null,
    "numero_rg": null,
    "prenom": "Adrien",
    "reason_extinction": null,
    "residence": null,
    "status": "Mesure en attente",
    "ti_id": null,
    "type": "Tutelle",
    "ville": null,
  },
]
`;
